src/libs/actions/OnyxUpdateManager.js

```diff
@@ -0,0 +1,78 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Log from '../Log';
+import * as SequentialQueue from '../Network/SequentialQueue';
+import * as App from './App';
+import * as OnyxUpdates from './OnyxUpdates';
+
+// This file is in charge of looking at the updateIDs coming from the server and comparing them to the last updateID that the client has.
+// If the client is behind the server, then we need to pause everything, get the missing updates from the server, apply those updates,
+// then restart everything. This will ensure that the client is up-to-date with the server and all the updates have been applied
+// in the correct order.
+// It's important that this file is separate and not imported by OnyxUpdates.js, so that there are no circular dependencies. Onyx
+// is used as a pub/sub mechanism to break out of the circular dependencies.
+// The circular dependency happen because this file calls the API GetMissingOnyxUpdates which uses the SaveResponseInOnyx.js file
+// (as a middleware). Therefore, SaveResponseInOnyx.js can't import and use this file directly.
```

Comment:
```suggestion
// This file is in charge of looking at the updateIDs coming from the server and comparing them to the last updateID that the client has.
// If the client is behind the server, then we need to 
// 1. Pause all sequential queue requests
// 2. Pause all Onyx updates from Pusher
// 3. Get the missing updates from the server
// 4. Apply those updates
// 5. Apply the original update that triggered this request (it could have come from either HTTPS or Pusher)
// 6. Restart the sequential queue
// 7. Restart the Onyx updates from Pusher
// This will ensure that the client is up-to-date with the server and all the updates have been applied in the correct order.
// It's important that this file is separate and not imported by OnyxUpdates.js, so that there are no circular dependencies. Onyx
// is used as a pub/sub mechanism to break out of the circular dependency.
// The circular dependency happens because this file calls API.GetMissingOnyxUpdates() which uses the SaveResponseInOnyx.js file
// (as a middleware). Therefore, SaveResponseInOnyx.js can't import and use this file directly.
```

Reply:
Done